improc/zogy.py

```diff
@@ -0,0 +1,286 @@
+# implement the ZOGY (Zackay, Ofek, & Gal-Yam 2016) image subtraction algorithm
+# ref: https://ui.adsabs.harvard.edu/abs/2016ApJ...830...27Z/abstract
+# also got some useful ideas from the implementation here: https://github.com/pmvreeswijk/ZOGY/blob/main/zogy.py#L15721
+
+import numpy as np
+import scipy
+
+
+def zogy_subtract(image_ref, image_new, psf_ref, psf_new, noise_ref, noise_new, flux_ref, flux_new, dx=None, dy=None):
+    """Perform ZOGY image subtraction.
+
+    This algorithm is based on Zackay et al. 2016 (https://ui.adsabs.harvard.edu/abs/2016ApJ...830...27Z/abstract).
+    It filters the new image with the PSF of the reference, and the reference image with the PSF of the new image
+    (which makes it symmetric for switching them) and divides by a normalization factor (in Fourier space)
+    that accounts for frequencies where there is less information.
+    This division makes sure the result is stable on all frequencies (no ringing as in deconvolution methods).
+
+    In addition to the subtracted image, it also calculates the subtraction PSF, and the "score" image,
+    which is the subtraction image match-filtered with its PSF. This image is normalized such that
+    any peak found in the score image has a value that is equivalent to the S/N of that detection.
+
+    Besides the classical score image (which is optimal assuming background dominated noise), the function also
+    returns the corrected score image, which takes into account the noise from bright sources.
+
+    The function also applies the astrometric noise correction when calculating score_corr, but only if
+    dx and dy are given (and if they are non-zero).
+    These values are... TODO: finish this
+
+    The last two outputs (alpha, alpha_std) represent a PSF-photometry measurement of the
```

Comment:
This is a good question. I think the math behind the algorithm will break down, but that doesn't mean it will necessarily fail or perform worse than deconvolutional methods.